src/bin/compileTS.js

```diff
@@ -0,0 +1,58 @@
+import { makeFlatSchemaId, makeFlatSchemaRef } from "@exabyte-io/esse.js/lib/js/esse/schemaUtils";
+import fs from "fs/promises";
+import { compile } from "json-schema-to-typescript";
+
+/**
+ * Compiles ESSE JSON schemas to TypeScript interfaces/types
+ * @param {Object} globalSchema
+ * @param {String} savePath
+ * @returns {Promise<void>}
+ * @example
+ * await compileTS(esseSchema, "./dist/types.ts");
+ */
+export async function compileTS(globalSchema, savePath) {
+    const preparedDefinitions = Object.entries(globalSchema.definitions).reduce(
+        (newDefinitions, [key, schema]) => {
+            if (schema.allOf && schema.properties) {
```

Comment:
Is separating `allOf` from `properties` necessary so that the TS interface is built correctly? I find this a bit difficult to follow, perhaps we could put a note in the docstring?

Reply:
Added a comment